src/js/providers/tracks-mixin.ts

```diff
@@ -310,8 +310,13 @@ const Tracks: TracksMixin = {
 
             for (i; i < len; i++) {
                 const track = tracks[i] as TextTrackLike;
-                let trackId: string = track._id || '';
-                if (!trackId) {
+                let trackId: string = track._id || '';                
+
+                if (hideIfOldTrackSafari(tracks, track, trackId)) {
```

Comment:
Maybe not worth calling if `trackId` is empty?

```suggestion
                if (trackId && hideIfOldTrackSafari(tracks, track, trackId)) {
```

Reply:
We still want to set track.parsed to true in that case, I forget what specific edge case it prevents but the idea is we want to know if any track has been through this function already.

Reply:
I'd like to call no matter what (trackId or not). This should be agnostic to how we index tracks, as the responsibility of this function is marking/checking if the track has been set by setTextTracks. Maybe I could rename to hasBeenSet but seemed long for a property name.